virttest/qemu_devices/qcontainer.py

```diff
@@ -2935,22 +2934,46 @@ def dimm_device_define_by_params(self, params, name):
 
     def memory_define_by_params(self, params, name):
         """
-        Create memory modules by params, include memory object and
-        pc-dimm devices.
+        Create memory modules by params, include memory object,
+        pc-dimm, nvdimm and virtio-mem devices.
         """
         params = params.object_params(name)
         devices = []
         if not self.has_device("pc-dimm"):
             LOG.warn("'PC-DIMM' does not support by your qemu")
             return devices
```

Comment:
Do we also need this now? Because we use `vm_memdev_model` to decide which one(dimm/virtio-mem) we want to use, it's so strange to check `pc-dimm` first and then attach a virtio-mem device. Another place need to be considered is https://github.com/avocado-framework/avocado-vt/blob/c7d21357b3a53295c0a5b819a1d1b81733e1796b/virttest/qemu_vm.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L1006-L1013

AFAICT, qemu support `pc-dimm` very early.

Reply:
Well, I agree with removing pc-dimm checks

Reply:
Changed!